### PR TITLE
gh action CI

### PR DESCRIPTION
**Pull request by deantchi**
this is just a skeleton CI that build and publishes Hord to docker cloud.
we can treat the PR as a WIP until the CI tasks is complete with testing and any additional CI tasks that may be needed before a release.

### CODE REVIEW

**Comment by CharlieC3** [comment]
@deantchi If this is still a WIP and not ready for review, can you change this to a draft?

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,17 @@
+FROM rust:bullseye as build
```
Comment (lgalabru) [inline Dockerfile]: We already have a similar Dockerfile in `dockerfiles/`

Comment (deantchi) [reply Dockerfile]: we should follow the same pattern as other repos and. that is to keep the `Dockerfile` at the root of the project.

**Comment by deantchi** [comment]
ready for review
recent run [here](https://github.com/hirosystems/ordhook/actions/runs/6042232339/job/16396878115)

File: Dockerfile (hunk 1 above)
Comment (lgalabru) [reply Dockerfile]: I like the idea of keeping the project root nice an clean, we will also probably end up with multiple dockerfiles (debian, alpine, etc). 
The `dockerfiles` directory convention is also adopted in our other repos (https://github.com/hirosystems/clarinet, https://github.com/hirosystems/chainhook).

**Review by lgalabru — CHANGES_REQUESTED** [review]
Looking great, thank you so much @deantchi! 
Could we please rely on the existing dockerfile instead?
Thanks!

**Comment by deantchi** [comment]
> Looking great, thank you so much @deantchi!
> Could we please rely on the existing dockerfile instead?
> Thanks!

done!

**Review by CharlieC3 — APPROVED** [review]
LGTM! :shipit:

**Comment by lgalabru** [comment]
Thanks @deantchi! 

Looking more closely at the build process, would it be possible to run in parallel:
- cargo test
- docker build

and only publish the image to docker hub if the test passed?
This simple trick would halve the CI time!

**Review by lgalabru — APPROVED** [review]

**Pull request merged**